Makefile

```diff
@@ -0,0 +1,113 @@
+# Package configuration
+PROJECT := gitbase-playground
+COMMANDS := cmd/gitbase-playground
+DOCKER_ORG := src-d
+DEPENDENCIES := \
+	github.com/golang/dep/cmd/dep \
+	github.com/jteeuwen/go-bindata \
+	github.com/golang/lint/golint
+GO_LINTABLE_PACKAGES := $(shell go list ./... | grep -v '/vendor/')
+GO_BUILD_ENV := CGO_ENABLED=0
+
+# Tools
+GODEP := dep
+GOLINT := golint
+GOVET := go vet
+BINDATA := go-bindata
+DIFF := diff
+GITADD := git add -A
+
+# Default rule
+all:
+
+# Including ci Makefile
+CI_REPOSITORY ?= https://github.com/dpordomingo/ci.git
+CI_BRANCH ?= v1
+CI_PATH ?= $(shell pwd)/.ci
+MAKEFILE := $(CI_PATH)/Makefile.main
+$(MAKEFILE):
+	@git clone --quiet --depth 1 -b $(CI_BRANCH) $(CI_REPOSITORY) $(CI_PATH);
+-include $(MAKEFILE)
+
+# Makefile.main::dependencies -> Makefile.main::$(DEPENDENCIES) -> this::dependencies
+dependencies: | front-dependencies back-dependencies exit
+
+# Makefile.main::test -> this::test
+test: front-test
+
+# this::build -> Makefile.main::build -> Makefile.main::$(COMMANDS)
+build: front-build back-build
+	@echo
+
+coverage: | test-coverage codecov
+
+lint: back-lint front-lint
+
+validate-commit: | \
+	back-dependencies \
+	back-ensure-assets-proxy \
+	front-fix-lint-errors \
+	no-changes-in-commit
+
+exit:
+	exit 0;
```

Comment:
where do you use it?

Reply:
Here: https://github.com/src-d/gitbase-playground/pull/15/files/98fc7ef6f847919a9611e5dfd90932885bc898b4#diff-b67911656ef5d18c4ae36cb6741b7965R33

Because the receipes from `Makefile.main::dependencies` must not be called

Reply:
I see. Could you please also add a comment. Makefiles are tricky and we don't work with them often, so I would really appreciate documentation for such non-standard cases.

Reply:
I'll do so.
`Makefile` &rarr; :boom: